### ISSUE TITLE
Incorrect sql with `group by` clause

### ISSUE THREAD

**Issue by YuxiLing** [PRQL/prql#3041]
### What happened?

prql version: 0.8.1

database: SQLite 3.42.0

comments: generated sql is incorrect in the following examples.

Error message from the database: HAVING clause on a non-aggregate query.

### PRQL input

```elm
from tb1
sort [c1]
aggregate [new_col = min c2]
filter (new_col >= 0)
```

### SQL output

```SQL
WITH table_3 AS (
  SELECT
    MIN(c2) AS new_col,
    c1
  FROM
    tb1
  ORDER BY
    c1
),
table_1 AS (
  SELECT
    new_col
  FROM
    table_3 AS table_2
)
SELECT
  new_col
FROM
  table_1 AS table_0
HAVING
  new_col >= 0

-- Generated by PRQL compiler version:0.8.1 (https://prql-lang.org)
```

### Expected SQL output

```SQL
WITH table_3 AS (
  SELECT
    MIN(c2) AS new_col,
    c1
  FROM
    tb1
  ORDER BY
    c1
),
table_1 AS (
  SELECT
    new_col
  FROM
    table_3 AS table_2
)
SELECT
  new_col
FROM
  table_1 AS table_0
WHERE
  new_col >= 0
```

### MVCE confirmation

- [X] Minimal example
- [X] New issue

**Comment by max-sixty** [comment]
I think this is fixed in 0.9.0 — please can you confirm @YuxiLing ?

```elm
from tb1
sort {c1}
aggregate {new_col = min c2}
filter (new_col >= 0)
```

```sql

WITH table_0 AS (
  SELECT
    MIN(c2) AS new_col,
    c1
  FROM
    tb1
  HAVING
    MIN(c2) >= 0
)
SELECT
  new_col
FROM
  table_0
```

**Comment by YuxiLing** [comment]
Yes, it is fixed in 0.9.0.
